src/v/kafka/protocol/schemata/generator.py

```diff
@@ -838,6 +876,32 @@ class response;
 {%- endif %}
 {%- endmacro %}
 
+{% macro field_flex_encoder(field, obj) %}
```

Comment:
This macro looks a lot like `field_encoder`, same with its cousin `field_decoder`. Would it make sense to break these out as strings and use string interpolation to change the different bits? Its like templates of templates...